src/main/kotlin/no/nav/syfo/oppfolgingsplanmottak/OppfolgingsplanApi.kt

```diff
@@ -1,36 +1,64 @@
 package no.nav.syfo.oppfolgingsplanmottak
 
-import io.ktor.server.application.*
-import io.ktor.server.auth.*
-import io.ktor.server.request.*
-import io.ktor.server.response.*
-import io.ktor.server.routing.*
+import io.ktor.server.application.call
+import io.ktor.server.auth.authenticate
+import io.ktor.server.request.receive
+import io.ktor.server.response.respond
+import io.ktor.server.routing.Routing
+import io.ktor.server.routing.get
+import io.ktor.server.routing.post
+import io.ktor.server.routing.route
+import java.sql.Timestamp
+import java.time.LocalDateTime
 import no.nav.syfo.application.api.auth.JwtIssuerType
 import no.nav.syfo.application.database.DatabaseInterface
+import no.nav.syfo.oppfolgingsplanmottak.database.findSendingStatus
 import no.nav.syfo.oppfolgingsplanmottak.database.storeFollowUpPlan
 import no.nav.syfo.oppfolgingsplanmottak.domain.FollowUpPlanDTO
-import no.nav.syfo.oppfolgingsplanmottak.domain.FollowUpPlanResponse
 import no.nav.syfo.util.getLpsOrgnumberFromClaims
 import no.nav.syfo.util.getOrgnumberFromClaims
+import org.slf4j.LoggerFactory
 import java.util.*
+import no.nav.syfo.oppfolgingsplanmottak.service.FollowUpPlanSendingService
+import no.nav.syfo.util.getSendingTimestamp
 
-fun Routing.registerOppfolgingsplanApi(
+fun Routing.registerFollowUpPlanApi(
     database: DatabaseInterface,
+    followUpPlanSendingService: FollowUpPlanSendingService,
 ) {
-    route("/api/v1/followupplan/write") {
+    val log = LoggerFactory.getLogger("FollowUpPlanApi")
+    val uuid = "uuid"
+
+    route("/api/v1/followupplan/") {
         authenticate(JwtIssuerType.MASKINPORTEN.name) {
-            post {
+            post("write") {
                 val followUpPlanDTO = call.receive<FollowUpPlanDTO>()
-                val uuid = UUID.randomUUID()
+                val planUuid = UUID.randomUUID()
+                val employerOrgnr = getOrgnumberFromClaims()
+                val lpsOrgnumber = getLpsOrgnumberFromClaims()
+
+                log.info("Received follow up plan from ${followUpPlanDTO.lpsName}, LPS orgnr: $lpsOrgnumber")
+
+                val followUpPlanResponse = followUpPlanSendingService.sendFollowUpPlan(followUpPlanDTO, planUuid, employerOrgnr)
+
+                val sentToGeneralPractitionerAt = getSendingTimestamp(followUpPlanResponse.isSentToGeneralPractitionerStatus)
 
                 database.storeFollowUpPlan(
-                    uuid = uuid,
+                    uuid = planUuid,
                     followUpPlanDTO = followUpPlanDTO,
-                    organizationNumber = getOrgnumberFromClaims(),
-                    lpsOrgnumber = getLpsOrgnumberFromClaims()
+                    organizationNumber = employerOrgnr,
+                    lpsOrgnumber = lpsOrgnumber,
+                    sentToGeneralPractitionerAt = sentToGeneralPractitionerAt,
+                    sentToNavAt = Timestamp.valueOf(LocalDateTime.now()),
```

Comment:
Bør vi sette sentToNavAt for kun de planene som har sendPlanToNav = true?

Ala `sentToNavAt = if (followUpPlanDTO.sendPlanToNav) Timestamp.valueOf(LocalDateTime.now()) else null`

Reply:
misforsto din forrige kommentar. fikser